src/http_handlers.cpp

```diff
@@ -0,0 +1,1347 @@
+      return true;
+    default:
+      return false;
+  }
+}
+
+// If a HTTP request maps directly to a Diameter
+// Server-Assignment-Type field, return the appropriate field.
+Cx::ServerAssignmentType ImpuRegDataTask::sar_type_for_request(RequestType type)
+{
+  switch (type)
+  {
+    case RequestType::DEREG_USER:
+      return Cx::ServerAssignmentType::USER_DEREGISTRATION;
+    case RequestType::DEREG_ADMIN:
+      return Cx::ServerAssignmentType::ADMINISTRATIVE_DEREGISTRATION;
+    case RequestType::DEREG_TIMEOUT:
+      return Cx::ServerAssignmentType::TIMEOUT_DEREGISTRATION;
+    case RequestType::DEREG_AUTH_FAIL:
+      return Cx::ServerAssignmentType::AUTHENTICATION_FAILURE;
+    case RequestType::DEREG_AUTH_TIMEOUT:
+      return Cx::ServerAssignmentType::AUTHENTICATION_TIMEOUT;
+    default:
+      // Should never be called for CALL or REG as they don't map to
+      // an obvious value.
+
+      // LCOV_EXCL_START
+      TRC_ERROR("Couldn't produce an appropriate SAR - internal software error'");
+      return Cx::ServerAssignmentType::ADMINISTRATIVE_DEREGISTRATION;
+      // LCOV_EXCL_STOP
+  }
+}
+
+ImpuRegDataTask::RequestType ImpuRegDataTask::request_type_from_body(std::string body)
+{
+  TRC_DEBUG("Determining request type from '%s'", body.c_str());
+  RequestType ret = RequestType::UNKNOWN;
+
+  std::string reqtype;
+  rapidjson::Document document;
+  document.Parse<0>(body.c_str());
+
+  if (!document.IsObject() || !document.HasMember("reqtype") || !document["reqtype"].IsString())
+  {
+    TRC_ERROR("Did not receive valid JSON with a 'reqtype' element");
+  }
+  else
+  {
+    reqtype = document["reqtype"].GetString();
+  }
+
+  if (reqtype == "reg")
+  {
+    ret = RequestType::REG;
+  }
+  else if (reqtype == "call")
+  {
+    ret = RequestType::CALL;
+  }
+  else if (reqtype == "dereg-user")
+  {
+    ret = RequestType::DEREG_USER;
+  }
+  else if (reqtype == "dereg-admin")
+  {
+    ret = RequestType::DEREG_ADMIN;
+  }
+  else if (reqtype == "dereg-timeout")
+  {
+    ret = RequestType::DEREG_TIMEOUT;
+  }
+  else if (reqtype == "dereg-auth-failed")
+  {
+    ret = RequestType::DEREG_AUTH_FAIL;
+  }
+  else if (reqtype == "dereg-auth-timeout")
+  {
+    ret = RequestType::DEREG_AUTH_TIMEOUT;
+  }
+
+  TRC_DEBUG("New value of _type is %d", ret);
+  return ret;
+}
+
+std::string ImpuRegDataTask::server_name_from_body(std::string body)
+{
+  rapidjson::Document document;
+  document.Parse<0>(body.c_str());
+
+  if (!document.IsObject() ||
+      !document.HasMember("server_name") ||
+      !document["server_name"].IsString())
+  {
+    TRC_DEBUG("Did not receive valid JSON with a 'server_name' element");
+    return "";
+  }
+  else
+  {
+    return document["server_name"].GetString();
+  }
+}
+
+std::string ImpuRegDataTask::wildcard_from_body(std::string body)
+{
+  rapidjson::Document document;
+  document.Parse<0>(body.c_str());
+
+  if (!document.IsObject() ||
+      !document.HasMember("wildcard_identity") ||
```

Comment:
maybe, this was just the existing code. Plus makes it easier to add trace I think